File: leetcode/twosum.js
let twoSum2 = function(nums, target) {
  let sorted = nums.sort((a,b) => a-b);
  console.log(sorted);
  let pointer1 = 0;
  let pointer2 = sorted.length - 1;

  for(let i = 0; i < sorted.length; i++) {
    let sum = sorted[pointer1] + sorted[pointer2];
    console.log(sum)
    if(sum === target)
      return [pointer1, pointer2];
    if(sum > target)
      pointer2--;
    else if(sum < target)
      pointer1++;
  }

  return [];
}

let twoSum = function(nums, target) {
  let map = new Map();
  for(let i = 0; i < nums.length; i++) {
    let mapValue = map.get(target - nums[i]);
    if(mapValue !== undefined)
     return [mapValue, i];
    else
      map.set(nums[i], i);
  }
}

console.log(twoSum([2,7,11,5], 9));

File: leetcode/leetbean/leetbean/src/main/java/org/strixsc/TwoSum.java
package org.strixsc;

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public class TwoSum {
    public static class Solution {
        public static int[] twoSum2P(int[] nums, int target)
        {
            // First start out by sorting the array;
            Arrays.sort(nums);

            int l = 0;
            int r = nums.length - 1;

            while (l != r)
            {
                int sum = nums[l] + nums[r];

                if (sum == target)
                {
                    return new int[]{l, r};
                }

                if (sum < target)
                {
                    l += 1;
                }

                if (sum > target)
                {
                    r -= 1;
                }
            }

            return new int[]{0, 0};
        }

        public static int[] twoSum(int[] nums, int target)
        {
            Map<Integer, Integer> hm = new HashMap<Integer, Integer>();

            for (int i = 0; i < nums.length; i++)
            {
                int lookingFor = target - nums[i];
                if (hm.containsKey(lookingFor)) {
                    return new int[]{i, hm.get(lookingFor)};
                }

                hm.put(nums[i], i);
            }

            return new int[]{0, 0};
        }
    }

    public static void main(String[] args) {
        System.out.println(Arrays.stream(Solution.twoSum(new int[]{-3, 4, 3, 90}, 0))
                .mapToObj(String::valueOf)
                .reduce((result, value) -> result + ", " + value)
                .orElse("")
        );
    }
}
